docs/word/word-add-ins-core-concepts.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Fundamental programming concepts with the Word JavaScript API
+description: Use the Word JavaScript API to build add-ins for Word.
+ms.date: 07/05/2019
+localization_priority: Priority
+---
+
+# Fundamental programming concepts with the Word JavaScript API
+
+This article describes concepts that are fundamental to using the [Word JavaScript API](../reference/overview/word-add-ins-reference-overview.md) to build add-ins for Word 2016 or later.
+
+## Asynchronous nature of Word APIs
+
+The Word JavaScript API is loaded by Office.js. The Word JavaScript API changes the way that you can interact with objects like documents and paragraphs. Rather than providing individual asynchronous APIs for retrieving and updating each of these objects, the Word JavaScript API provides "proxy" JavaScript objects that correspond to the real objects running in Word. You can interact with these proxy objects by synchronously reading and writing their properties and calling synchronous methods to perform operations on them. These interactions with proxy objects aren't immediately realized in the running script. The **context.sync** method synchronizes the state between your running JavaScript and the real objects in Office by executing queued instructions and retrieving properties of loaded Word objects for use in your script.
+
+## Referencing Office.js
+
+You can reference Office.js from the following locations:
+
+- `https://appsforoffice.microsoft.com/lib/1/hosted/office.js` - use this resource for production add-ins.
+
+- `https://appsforoffice.microsoft.com/lib/beta/hosted/office.js` - use this resource to try out preview features.
+
+## Word JavaScript API requirement sets
+
+Requirement sets are named groups of API members. Office Add-ins use requirement sets specified in the manifest or use a runtime check to determine whether an Office host supports APIs that an add-in needs. For detailed information about Word JavaScript API requirement sets, see [Word JavaScript API requirement sets](../reference/requirement-sets/word-api-requirement-sets.md).
+
+## Running Word add-ins
+
+To run your add-in, use an **Office.initialize** event handler. For more information about add-in initialization, see [Understanding the API](/office/dev/add-ins/develop/understanding-the-javascript-api-for-office).
+
+Add-ins that target Word 2016 or later run by passing a function into the **Word.run()** method. The function passed into the **run** method must have a context argument. This [context object](/javascript/api/word/word.requestcontext) is different than the context object you get from the Office object, but it is also used to interact with the Word runtime environment. The context object provides access to the Word JavaScript API object model. The following example shows how to initialize and run a Word add-in by using the **Word.run()** method.
+
+```js
+(function () {
+    "use strict";
```

Comment:
Why is this "use strict" line here? Just curious for my own learning.

Reply:
Good question. Here's a decent explanation of that that does: https://www.w3schools.com/js/js_strict.asp. At some point in the past, this line appeared in almost all of our code samples (and maybe even in our SL snippets?). Ricky may be able to provide more insight about the rationale for including it.